### ISSUE TITLE
Main and related sub-zaak need additional relation with Afleidingswijze=gerelateerde or hoofdzaak

### ISSUE THREAD

**Issue by hsiliev** [open-zaak/open-zaak#1839]
### Product versie / Product version

1.15.0

### Omschrijf het probleem / Describe the bug

We have a main and sub-zaak (HOOFDZAAK and DEELZAAK). When trying to use result type with `Afleidingswijze` set to either `gerelateerde zaak` or `hoofdzaak` we see that these main-sub zaak relation is not taken into account and instead we get this error: `Geen gerelateerde zaken aan zaak gekoppeld om brondatum uit af te leiden.`

When I add additional relation in `Relevante zaak relaties` in OpenZaak admin between the two ZAAKs the above procedure works. However in this way we're changing the type of the relation between the two ZAAKs and this seems wrong to do.

### Stappen om te reproduceren / Steps to reproduce

1. Create main zaak
2. Create sub-zaak 
3. Close the main zaak with resulttype having `Afleidingswijze` =  `gerelateerde zaak` or `hoofdzaak` 

### Verwacht gedrag / Expected behavior

Main and sub-zaak relation is honoured and used to determine the end-date as described in https://vng-realisatie.github.io/gemma-zaken/themas/achtergronddocumentatie/archiveren

**Comment by joeribekker** [comment]
So, we need to determine if, based on the Zaaktype->Resultaattype "afleidingswijze" (set to main or related case) we properly determine the archive-action date (Archiefactiedatum) based on the main or reltaed case.

Can we add a test for this to prove/disprove this works or not and or fix.

**Comment by hsiliev** [comment]
Yep, we have logic in ZAC that does not allow closing of parent zaak if child zaak is not closed. 

But even so that does not really work with OpenZaak. Seems like hoofdzaak and deelzaak relations does not mean much here (or are just ignored). 

The only way to make such relation work was to add another type of relation between the hoofdzaak and deelzaak. This however seems incorrect.

**Comment by Hugo-ter-Doest** [comment]
> Yep, we have logic in ZAC that does not allow closing of parent zaak if child zaak is not closed.

Imho this should be logic of ZGW API's to safeguard this.

Hoofdzaak/deelzaak is a stronger relationship than gerelateerde zaken. So using another type of relation is nog advisable.

**Comment by joeribekker** [comment]
We need to work out some more business rules for this topic. But, we do think that Open Zaak does indeed not have any logic related to deelzaak/hoofdzaak implemented.

@annashamray can you confirm?

**Comment by annashamray** [comment]
@joeribekker 
We have following validation for hoofdzaak/deelzaken:
* deelzaken can't become hoofdzaken for other zaken
* deelzaken and hoofdzaken should have zaaktypen with the same relation

But there is no validation for closed zaken and we don't close deelzaken automatically if the hoofdzaken is closed

**Comment by joeribekker** [comment]
Refinement: Hugo suggests that all deelzaken need to be closed, before the hoofdzaak can be closed (validation thing). Once the hoofdzaak is closed, we could re-evaluate the archiving parameters (afleidingswijze=hoofdzaak, we need to do something).

@Hugo-ter-Doest will validate this behaviour with the IM group.

**Comment by Hugo-ter-Doest** [comment]
I sent a message to the Werkgroep Informatiebeheer of Dimpact with some questions to validate our approach. To be continued.
